aiven/client/client.py

```diff
@@ -99,56 +98,53 @@ def _execute(self, func, method, path, body, params=None):
 
         self.http_log.debug("-----Response End-----")
 
-        if not str(response.status_code).startswith("2"):
-            raise Error(response, status=response.status_code)
+    def verify(
+        self,
+        op: str,
+        path: str,
+        body: Union[Dict, Text] = None,
+        params: Dict = None,
+        result_key: str = None,
+        retry: Union[Retry, None, int] = None
+    ) -> Dict:
+        path = self.api_prefix + path
 
-        return response
+        if retry is not None:
+            if isinstance(retry, Retry):
+                attempts = 1 + (retry or 0)
+                retry = Retry.from_int(attempts)
 
-    def get(self, path="", params=None):
-        """HTTP GET"""
-        return self._execute(self.session.get, "GET", path, body=None, params=params)
+            # If `retry` was explicit provided assume the operation is
+            # idempotent regardless of the verb.
+            retry.new(method_whitelist=[op])
 
-    def post(self, path="", body=None, params=None):
-        """HTTP POST"""
-        return self._execute(self.session.post, "POST", path, body, params)
+            # Retry is only configurable at the session level, so a new session
+            # is necessary.
+            session = session_with_retries(retry)
+        else:
+            session = self.session
+
+        url = self.base_url + path
 
-    def put(self, path="", body=None, params=None):
-        """HTTP PUT"""
-        return self._execute(self.session.put, "PUT", path, body, params)
+        headers = dict()
+        if self.auth_token:
+            headers["authorization"] = "aivenv1 {token}".format(token=self.auth_token)
 
-    def delete(self, path="", body=None, params=None):
-        """HTTP DELETE"""
-        return self._execute(self.session.delete, "DELETE", path, body, params)
+        data = None
+        jsondata: Optional[Dict] = None
 
-    def verify(self, op, path, body=None, params=None, result_key=None, retry=None):
-        # Retry GET operations by default
-        if retry is None and op == self.get:  # pylint: disable=comparison-with-callable
-            attempts = 3
+        if isinstance(body, dict):
+            jsondata = body
         else:
-            attempts = 1 + (retry or 0)
```

Comment:
I don't understand why the number of retries is increased by one if provided, to me it would make more sense to have `retry or 1` here